### PR TITLE
[macosx] set minimum osx version to 10.13

### PR DESCRIPTION
**Pull request by phunkyfish**
## Description

Set minimum supported OSX version to OSX High Sierra

## Motivation and Context

OSX High Sierra runs on Apple devices since from 2009 which is plenty of coverage.

Main motivation was I had to create a patch because gnutls now assume functions available since high sierra are now available by default.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [X] **None of the above** (please explain below)

Not quite sure how to categorise this.

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [X] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by phunkyfish** [comment]
Also note that version prior to 10.12 no longer receive security updates as far as I can tell.

File: docs/README.macOS.md [hunk 1]
```diff
@@ -9,12 +9,12 @@ This guide has been tested with macOS 10.13.4()17E199 High Sierra and Xcode 9.3(
 3. **[Get the source code](#3-get-the-source-code)**
 4. **[Configure and build tools and dependencies](#4-configure-and-build-tools-and-dependencies)**
 5. **[Build binary add-ons](#5-build-binary-add-ons)**
-6. **[Build Kodi](#6-build-kodi)**  
-  6.1. **[Build with Xcode](#61-build-with-xcode)**  
-  6.2. **[Build with xcodebuild](#62-build-with-xcodebuild)**  
+6. **[Build Kodi](#6-build-kodi)**
```
Comment (fuzzard) [inline docs/README.macOS.md]: Removing the double spaces is wrong. Its the markdown for a newline. Id revert any of these changes

**Comment by fuzzard** [comment]
There is also still a reference to 10.9 in the Prequisites section of the readme

```
Device with OSX 10.9 or newer to run Kodi after build.
```

File: docs/README.macOS.md (hunk 1 above)
Comment (phunkyfish) [reply docs/README.macOS.md]: Ah, ok, did not realise. Will fix

**Review by Rechi — CHANGES_REQUESTED** [review]
There is still code which exists only for 10.9 - 10.11. That one has to be removed too

**Comment by fuzzard** [comment]
https://github.com/xbmc/xbmc/blob/ad19c674e8022f9df778bad3b37a564b8d5dfc2d/xbmc/platform/darwin/osx/smc.c#L92

I'll see what else I can find phunky

**Comment by fuzzard** [comment]
https://github.com/xbmc/xbmc/blob/ad19c674e8022f9df778bad3b37a564b8d5dfc2d/xbmc/rendering/gl/RenderSystemGL.cpp#L404

Looks specific to some old gfx hardware

**Comment by phunkyfish** [comment]
Do I need to be concerned with `MAC_OS_X_VERSION_10_5`?

Or just:

- `MAC_OS_X_VERSION_10_9`
- `MAC_OS_X_VERSION_10_10`
- `MAC_OS_X_VERSION_10_11`

???

**Comment by phunkyfish** [comment]
> https://github.com/xbmc/xbmc/blob/ad19c674e8022f9df778bad3b37a564b8d5dfc2d/xbmc/rendering/gl/RenderSystemGL.cpp#L404
> 
> Looks specific to some old gfx hardware

Thing is though, how old would that hardware be. If it can still run Sierra it's ok. Hard to tell.

**Comment by kambala-decapitator** [comment]
`$ grep -rn -e "NSFoundationVersionNumber" -e "respondsToSelector:" -e "MAC_OS_X_VERSION_" *`

lines in `tools/EventClients/Clients/OSXRemote/HIDRemote/HIDRemote.m`:
- 266
- 637
- 747
- 2087

also check the above file together with its header wrt `MAC_OS_X_VERSION_*`. Now you have `MAC_OS_X_VERSION_MIN_REQUIRED` set to 10.12 and `MAC_OS_X_VERSION_MAX_ALLOWED` is set to 10.14.

more stuff:
- xbmc/utils/SystemInfo.cpp:1253

**Comment by fuzzard** [comment]
ATI 9200 was from 2003, the ATI 1600 from 2009. If your culling < 10.12, im sure they will be ok to remove as well

**Comment by phunkyfish** [comment]
There is an awful lot of `MAC_OS_X_VERSION_10_5` and other versions. I assume I can take anything <= 10_12 and just assume it's ok as TARGET_DARWIN_OSX.

File: tools/EventClients/Clients/OSXRemote/HIDRemote/HIDRemote.m [hunk 2]
```diff
@@ -633,8 +601,7 @@ - (BOOL)isStopping
 #pragma mark - PRIVATE: Application becomes active / inactive handling for kHIDRemoteModeExclusiveAuto
 - (void)_appStatusChanged:(NSNotification *)notification
 {
-	#if HIDREMOTE_THREADSAFETY_HARDENED_NOTIFICATION_HANDLING
-	if ([self respondsToSelector:@selector(performSelector:onThread:withObject:waitUntilDone:)]) // OS X 10.5+ only
+	if ([self respondsToSelector:@selector(performSelector:onThread:withObject:waitUntilDone:)])
```
Comment (kambala-decapitator) [inline tools/EventClients/Clients/OSXRemote/HIDRemote/HIDRemote.m]: it always evaluates to `true`, drop the check

File: tools/EventClients/Clients/OSXRemote/HIDRemote/HIDRemote.m [hunk 3]
```diff
@@ -743,16 +709,14 @@ - (void)_handleNotifications:(NSNotification *)notification
 {
 	NSString *notificationName;
 
-	#if HIDREMOTE_THREADSAFETY_HARDENED_NOTIFICATION_HANDLING
-	if ([self respondsToSelector:@selector(performSelector:onThread:withObject:waitUntilDone:)]) // OS X 10.5+ only
+	if ([self respondsToSelector:@selector(performSelector:onThread:withObject:waitUntilDone:)])
```
Comment (kambala-decapitator) [inline tools/EventClients/Clients/OSXRemote/HIDRemote/HIDRemote.m]: it always evaluates to `true`, drop the check

File: tools/EventClients/Clients/OSXRemote/HIDRemote/HIDRemote.m [hunk 4]
```diff
@@ -2076,26 +2040,16 @@ - (void)_secureInputNotificationFor:(io_service_t)service messageType:(natural_t
 
 - (void)_computerDidWake:(NSNotification *)aNotification
 {
-	// Work around for a bug in 10.8, where exclusive connections may be degraded to shared connections after a sleep/wakeup cycle (credit: Paul Duggan from Galaxy Software)
-	#ifdef NSAppKitVersionNumber10_8
-	if (NSAppKitVersionNumber >= NSAppKitVersionNumber10_8)
-	#else
-	if (NSAppKitVersionNumber >= 1187)
-	#endif
+	if ([self respondsToSelector:@selector(performSelector:onThread:withObject:waitUntilDone:)])
```
Comment (kambala-decapitator) [inline tools/EventClients/Clients/OSXRemote/HIDRemote/HIDRemote.m]: it always evaluates to `true`, drop the check

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]

**Comment by phunkyfish** [comment]
Should be ok now @kambala-decapitator

**Comment by Rechi** [comment]
Code at lib/libUPnP and tools/EventClients/Clients/OSXRemote/HIDRemote is 3rd party code and should be kept as is.

**Comment by phunkyfish** [comment]
Anything I might have missed @Rechi ?

**Comment by Rechi** [comment]
Yes, the new minimum system version isn't update in `Info.plist` and there is still code in tools/depends which isn't needed anymore.

**Comment by phunkyfish** [comment]
Ah yes, I missed archsetup for OS X which should cover the plist.

I will need your guidance on tools/depends as I’m not sure what I’m looking for.

**Comment by Rechi** [comment]
`git grep -E '10\.(9|1[0-2])[^0-9]' -- tools/depends/`

**Comment by phunkyfish** [comment]
Cool, I also removed that code from darwin_embedded which should be ok right?

**Comment by phunkyfish** [comment]
Should I add a comment to the remaining functions that are only available in 10.13?

**Comment by phunkyfish** [comment]
jenkins build this please

**Comment by phunkyfish** [comment]
Ouch, didn't pass on single Apple platform.

**Comment by phunkyfish** [comment]
jenkins build this please

**Comment by phunkyfish** [comment]
@kambala-decapitator for OSX it builds but there are some deprecated functions:
```
01:12:55 /Users/Shared/jenkins/workspace/OSX-64/xbmc/platform/darwin/osx/SDLMain.mm:88:43: warning: 'NSAlternateKeyMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
01:12:55   [menuItem setKeyEquivalentModifierMask:(NSAlternateKeyMask|NSCommandKeyMask)];
01:12:55                                           ^~~~~~~~~~~~~~~~~~
01:12:55                                           NSEventModifierFlagOption
01:12:55 /Applications/Xcode10.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSEvent.h:184:35: note: 'NSAlternateKeyMask' has been explicitly marked deprecated here
01:12:55 static const NSEventModifierFlags NSAlternateKeyMask          NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSEventModifierFlagOption", 10_0, 10_12) = NSEventModifierFlagOption;
01:12:55                                   ^
01:12:55 /Users/Shared/jenkins/workspace/OSX-64/xbmc/platform/darwin/osx/SDLMain.mm:88:62: warning: 'NSCommandKeyMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
01:12:55   [menuItem setKeyEquivalentModifierMask:(NSAlternateKeyMask|NSCommandKeyMask)];
01:12:55                                                              ^~~~~~~~~~~~~~~~
01:12:55                                                              NSEventModifierFlagCommand
01:12:55 /Applications/Xcode10.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSEvent.h:185:35: note: 'NSCommandKeyMask' has been explicitly marked deprecated here
01:12:55 static const NSEventModifierFlags NSCommandKeyMask            NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSEventModifierFlagCommand", 10_0, 10_12) = NSEventModifierFlagCommand;
01:12:55                                   ^
01:12:55 /Users/Shared/jenkins/workspace/OSX-64/xbmc/platform/darwin/osx/SDLMain.mm:194:46: warning: 'NSTitledWindowMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
01:12:55   [window setStyleMask: [window styleMask] ^ NSTitledWindowMask ];
01:12:55                                              ^~~~~~~~~~~~~~~~~~
01:12:55                                              NSWindowStyleMaskTitled
01:12:55 
/Applications/Xcode10.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSWindow.h:952:32: note: 'NSTitledWindowMask' has been explicitly marked deprecated here
01:12:55 static const NSWindowStyleMask NSTitledWindowMask NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSWindowStyleMaskTitled", 10_0, 10_12) = NSWindowStyleMaskTitled;
01:12:55                                ^
01:12:55 /Users/Shared/jenkins/workspace/OSX-64/xbmc/platform/darwin/osx/SDLMain.mm:195:37: warning: 'NSTitledWindowMask' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
01:12:55   BOOL isSet = [window styleMask] & NSTitledWindowMask;
01:12:55                                     ^~~~~~~~~~~~~~~~~~
01:12:55                                     NSWindowStyleMaskTitled
01:12:55 /Applications/Xcode10.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSWindow.h:952:32: note: 'NSTitledWindowMask' has been explicitly marked deprecated here
01:12:55 static const NSWindowStyleMask NSTitledWindowMask NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSWindowStyleMaskTitled", 10_0, 10_12) = NSWindowStyleMaskTitled;
01:12:55                                ^
01:12:55 /Users/Shared/jenkins/workspace/OSX-64/xbmc/platform/darwin/osx/SDLMain.mm:334:49: warning: 'NSApplicationDefined' is deprecated: first deprecated in macOS 10.12 [-Wdeprecated-declarations]
01:12:55   NSEvent* event = [NSEvent otherEventWithType: NSApplicationDefined
01:12:55                                                 ^~~~~~~~~~~~~~~~~~~~
01:12:55                                                 NSEventTypeApplicationDefined
01:12:55 /Applications/Xcode10.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSEvent.h:79:26: note: 'NSApplicationDefined' has been explicitly marked deprecated here
01:12:55 static const NSEventType NSApplicationDefined   NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSEventTypeApplicationDefined", 10_0, 10_12) = NSEventTypeApplicationDefined;
01:12:55                          ^
01:12:55 /Users/Shared/jenkins/workspace/OSX-64/xbmc/platform/darwin/osx/SDLMain.mm:559:20: warning: null passed to a callee that requires a non-null argument [-Wnonnull]
01:12:55     [xbmc_delegate applicationWillTerminate:NULL];
01:12:55                    ^                        ~~~~
01:12:55 6 warnings generated.
```

Can you help?

**Comment by phunkyfish** [comment]
I'm not really sure what is wrong with iOS/tvOS. Unless I shouldn't have removed

```
if test "@platform_os@" = "darwin_embedded"; then
  # Xcode 8 + 10.11, clock_gettime getentropy is present
  # in 10.12 but will get wrongly detected if building on 10.11
  ac_cv_search_clock_gettime=no
  ac_cv_func_clock_gettime=no
  ac_cv_func_getentropy=no

```

**Comment by Rechi** [comment]
What do you think are the implications of removing those lines? (hint: ignore the comment)

**Comment by phunkyfish** [comment]
So they are not related to the osx min version change. Instead bound to ios/tvos versions I guess. I'll revert those now and I guess remove the comment as it's misleading.

**Comment by Rechi** [comment]
How will the following work, with the current changes in this PR?
https://github.com/xbmc/xbmc/blob/1cf5d5e47f33baba6e6344a0c9f39a3689d18cc4/docs/README.iOS.md#L58

**Comment by phunkyfish** [comment]
> Xcode 8.x against iOS SDK 10.x on 10.11.x (El Capitan)(recommended)

Will remove

**Comment by phunkyfish** [comment]
> @kambala-decapitator for OSX it builds but there are some deprecated functions:
> 
> ```
> 01:12:55                          ^
> 01:12:55 /Users/Shared/jenkins/workspace/OSX-64/xbmc/platform/darwin/osx/SDLMain.mm:559:20: warning: null passed to a callee that requires a non-null argument [-Wnonnull]
> 01:12:55     [xbmc_delegate applicationWillTerminate:NULL];
> 01:12:55                    ^                        ~~~~
> 01:12:55 6 warnings generated.
> ```
> 
> Can you help?

Ok, the first 5 warnings were obvious, not sure about number 6 though.

**Comment by phunkyfish** [comment]
> > @kambala-decapitator for OSX it builds but there are some deprecated functions:
> > ```
> > 01:12:55                          ^
> > 01:12:55 /Users/Shared/jenkins/workspace/OSX-64/xbmc/platform/darwin/osx/SDLMain.mm:559:20: warning: null passed to a callee that requires a non-null argument [-Wnonnull]
> > 01:12:55     [xbmc_delegate applicationWillTerminate:NULL];
> > 01:12:55                    ^                        ~~~~
> > 01:12:55 6 warnings generated.
> > ```
> > 
> > 
> > Can you help?
> 
> Ok, the first 5 warnings were obvious, not sure about number 6 though.

Ok, I think I fixed this one too.

File: xbmc/platform/darwin/osx/SDLMain.mm [hunk 5]
```diff
@@ -556,7 +556,8 @@ int main(int argc, char *argv[])
     status = SDL_main(gArgc, gArgv);
     SDL_Quit();
 
-    [xbmc_delegate applicationWillTerminate:NULL];
```
Comment (kambala-decapitator) [inline xbmc/platform/darwin/osx/SDLMain.mm]: this is an old warning and can be ignored in this PR

Comment (phunkyfish) [reply xbmc/platform/darwin/osx/SDLMain.mm]: So then the question will be where is the 6th new warning as I only found 5!

Comment (phunkyfish) [reply xbmc/platform/darwin/osx/SDLMain.mm]: Although this makes 6 so Jenkins is happy now.

Comment (kambala-decapitator) [reply xbmc/platform/darwin/osx/SDLMain.mm]: I think is the last one: https://jenkins.kodi.tv/job/OSX-64/16848/warnings2Result/


ThreadImpl.cpp:65, kodi compiler warnings, Priority: Normal

'syscall' is deprecated: first deprecated in macOS 10.12 - syscall(2) is unsupported; please switch to a supported interface. For SYS_kdebug_trace use kdebug_signpost(). [-Wdeprecated-declarations]

File: xbmc/threads/platform/pthreads/ThreadImpl.cpp [hunk 6]
```diff
@@ -61,6 +61,10 @@ static pid_t GetCurrentThreadPid_()
   return gettid();
 #elif TARGET_DARWIN_TVOS
   return pthread_mach_thread_np(pthread_self());
+#elif defined(__APPLE__) && defined(__MACH__)
```
Comment (kambala-decapitator) [inline xbmc/threads/platform/pthreads/ThreadImpl.cpp]: why not using the same as tvOS?

Comment (phunkyfish) [reply xbmc/threads/platform/pthreads/ThreadImpl.cpp]: Because it was not the the first thing google suggested as a replacement ;)

Comment (phunkyfish) [reply xbmc/threads/platform/pthreads/ThreadImpl.cpp]: But much cleaner, thanks ;)

**Comment by phunkyfish** [comment]
Ok, should be good now.

**Comment by phunkyfish** [comment]
@Rechi @kambala-decapitator are you happy with this now?

File: xbmc/threads/platform/pthreads/ThreadImpl.cpp [hunk 7]
```diff
@@ -61,6 +61,8 @@ static pid_t GetCurrentThreadPid_()
   return gettid();
 #elif TARGET_DARWIN_TVOS
```
Comment (kambala-decapitator) [inline xbmc/threads/platform/pthreads/ThreadImpl.cpp]: I'd just use this for all Darwin targets:

```suggestion
 #elif TARGET_DARWIN
```

File: xbmc/platform/darwin/osx/SDLMain.mm (hunk 5 above)
Comment (kambala-decapitator) [inline xbmc/platform/darwin/osx/SDLMain.mm]: so, why change this? The warning didn't appear because of deployment target raise, it had been there before.

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]

File: xbmc/threads/platform/pthreads/ThreadImpl.cpp (hunk 7 above)
Comment (phunkyfish) [reply xbmc/threads/platform/pthreads/ThreadImpl.cpp]: Ok

File: xbmc/platform/darwin/osx/SDLMain.mm (hunk 5 above)
Comment (phunkyfish) [reply xbmc/platform/darwin/osx/SDLMain.mm]: I figured one less warning was better 😉

Comment (kambala-decapitator) [reply xbmc/platform/darwin/osx/SDLMain.mm]: well, creating a fake notification just to silence warning isn't a proper fix :) You'd rather create a new method that will be called by this code and by `-applicationWillTerminate:`. But this really looks unrelated to the PR.

Comment (phunkyfish) [reply xbmc/platform/darwin/osx/SDLMain.mm]: Fair, will remove

**Comment by Rechi** [comment]
There are still dependencies which explicitly disable `clock_gettime` and `getentropy` availability for osx.
Also splitting the minimum os version bump and the function availability hints is wrong.

**Comment by phunkyfish** [comment]
I can squash it to one commit once everyone is happy with it.

@Rechi do you mean the patches as part of `tools/depends/target` that need to be updated to exclude OS X?

**Comment by phunkyfish** [comment]
Anything in addition to the following that needs to be updated:

- https://github.com/xbmc/xbmc/blob/3c3bb43feac3da359e83d37919aca83a6c2fbc5f/tools/depends/target/curl/Makefile#L25-L27
- https://github.com/xbmc/xbmc/blob/3c3bb43feac3da359e83d37919aca83a6c2fbc5f/tools/depends/target/samba-gplv3/Makefile#L54-L61
- Remove gnutls getentropy patch

**Comment by phunkyfish** [comment]
jenkins build this please

File: tools/depends/target/curl/Makefile [hunk 8]
```diff
@@ -22,7 +22,7 @@ $(TARBALLS_LOCATION)/$(ARCHIVE):
 $(PLATFORM): $(TARBALLS_LOCATION)/$(ARCHIVE) $(DEPS)
 	rm -rf $(PLATFORM)/*; mkdir -p $(PLATFORM)
 	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE)
-ifeq (darwin, $(findstring darwin, $(HOST)))
+ifeq ($(OS), darwin_embedded)
```
Comment (Rechi) [inline tools/depends/target/curl/Makefile]: Isn't this only required for iphoneos?

Comment (phunkyfish) [reply tools/depends/target/curl/Makefile]: Ya, I spotted that in the samba-gplv3 dependency but was not sure if it was the same. I guess it must be. Will change shortly.

Comment (phunkyfish) [reply tools/depends/target/curl/Makefile]: Done

**Comment by Rechi** [comment]
> Also note that version prior to 10.12 no longer receive security updates as far as I can tell.

Any reason to not drop 10.12? From my understanding versions prior 10.13 don't get security updates.

**Comment by phunkyfish** [comment]
> > Also note that version prior to 10.12 no longer receive security updates as far as I can tell.
> 
> Any reason to not drop 10.12? From my understanding versions prior 10.13 don't get security updates.

No reason, will make that change later today.

I could make the same argument for ios, any reason we would not move up to iOS 11? That would mean no more crazy patching of functions.

**Comment by phunkyfish** [comment]
@kambala-decapitator what are your thoughts on iOS to 11.0?

**Comment by fuzzard** [comment]
Go nuts if you think you want to. If you do though, that's the end of 32bit iOS platform. iOS 11 is 64bit only, and iOS 10 was the last to support 32 bit devices (iPhone 5/5c, iPad 4)

**Comment by phunkyfish** [comment]
Hmmm, I wonder is it too late in the release cycle to drop those.

**Comment by fuzzard** [comment]
Has been no alpha release yet, never too late at this stage

**Comment by phunkyfish** [comment]
True

**Comment by phunkyfish** [comment]
I will merge this tomorrow if there are no further objections (and Jenkins is happy).

**Comment by kambala-decapitator** [comment]
I'm fine with raising iOS deployment target, although not sure if it's really necessary

**Comment by Rechi** [comment]
There is still a dependency which explicitly disables `futimens` and `utimensat` availability for osx.
Sure go ahead with an iOS deployment target bump, as long as you drop all the outdated code.

**Comment by phunkyfish** [comment]
Ah, I had forgot to check depends/native.

**Comment by phunkyfish** [comment]
Anything else @Rechi

**Review by Rechi — APPROVED** [review]

**Review by kambala-decapitator — APPROVED** [review]

**Pull request merged**